lib/mysql.js

```diff
@@ -249,6 +240,28 @@ MySQL.prototype.updateOrCreate = MySQL.prototype.save =
       callback(err, data, meta);
     });
   };
+  
+/**
+ * Replace if the model instance exists with the same id or create a new instance
+ *
+ * @param {String} model The model name
+ * @param {Object} data The model instance data
+ * @param {Function} [callback] The callback function
+ */
+MySQL.prototype.replaceOrCreate = function (model, data, options, callback) {
+  this.modifyOrCreate(model, data, options, 'buildReplaceFields', callback);
```

Comment:
So there are two ways:

1) use the function name as you suggested, I think it's less clean because you cannot pass an anonymous function

2) pass a wrapper function that preserves `this`. You could use `this.buildReplaceFields.bind(this)`, except that `bind` has terrible performance. Therefore this would be a more performant solution:

```js
var self = this;
this.modifyOrCreate(model, data, options, function(model, data) { self.buildReplaceFields(model, data); }, callback);
```